deploy/lib/deployContainers.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const BbPromise = require('bluebird');
+
+module.exports = {
+  deployContainers() {
+    this.serverless.cli.log('Deploying Containers...');
+    return BbPromise.bind(this)
+      .then(this.deployEachContainer)
+      .then(() => this.serverless.cli.log('Waiting for container deployments, this may take multiple minutes...'))
+      .catch((err) => {
+        throw new Error(err.response.data.message)
+      })
+      .then(this.waitContainersAreDeployed);
+  },
+
+  deployEachContainer() {
+    const promises = this.containers.map(
+      cont => this.provider.apiManager.post(`containers/${cont.id}/deploy`, {})
+      .then(response => response.data)
+      .catch((err) => {
+        throw new Error(err.response.data.message)
+      }),
+    );
+
+    return Promise.all(promises);
+  },
+
+  waitContainersAreDeployed() {
+    return this.provider.apiManager.get(`namespaces/${this.namespace.id}/containers`)
+      .then((response) => {
+        const containers = response.data.containers || [];
+        let containersAreReady = true;
+        for (let i = 0; i < containers.length; i += 1) {
+          const cont = response.data.containers[i];
+          if (cont.status === 'error') {
```

Comment:
I believe you should create an object `{[container_name]: [error_message]}`, this way if there are multiple errors (if I deploy multiple containers, which encounter errors) you can format a message to point out every error message linked to its container.

Reply:
We should change the way we treat errors. Today it is throwing an error and stops the plugin execution.